src/transpiler.ts

```diff
@@ -13,7 +13,51 @@ class Transpiler {
             throw new Error('The generated WebAssembly is invalid');
         }
 
-        return new WebAssembly.Instance(new WebAssembly.Module(module.emitBinary()), {}).exports;
+        const wasmModule = new WebAssembly.Module(module.emitBinary());
+
+        return (functionName: string, ...params: any[]) => {
+            let wasmParams = params;
+            let importObject = {};
+
+            if (params.some((parameter) => parameter instanceof Array)) {
+                const memory = new WebAssembly.Memory({ initial: this.calculateInitialMemorySize(params) });
+                const writeableMemory = new Uint32Array(memory.buffer);
+
+                wasmParams = this.fillMemory(wasmParams, writeableMemory);
+                importObject = { transpilerImports: { memory } };
+            }
+
+            return new WebAssembly.Instance(wasmModule, importObject).exports[functionName](...wasmParams);
+        };
+    }
+
+    private static fillMemory(params: any[], writeableMemory: Uint32Array) {
+        // A copy of the params array is needed, to not override the content of the original one
+        const wasmParams = params.concat();
```

Comment:
Add a test to check if the parameters were not modified

Reply:
If no `concat()` is used, it would only modify the top level `params` array elements. The passed array does not get modified. Therefore it can not be tested from the outside. However, the reference to the passed array would be lost to us, making it impossible to later on write back the WebAssembly memory values.